android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt

```diff
@@ -110,18 +112,32 @@ class FlutterInboxMessageView(private var context: Context, channel: MethodChann
     }
 }
 
-class AirshipPlugin : MethodCallHandler {
+class AirshipPlugin : MethodCallHandler, FlutterPlugin {
+
+    private lateinit var channel : MethodChannel
+
+    private lateinit var context: Context
 
     companion object {
         @JvmStatic
         fun registerWith(registrar: Registrar) {
             val channel = MethodChannel(registrar.messenger(), "com.airship.flutter/airship")
             channel.setMethodCallHandler(AirshipPlugin())
             EventManager.shared.register(registrar)
-            registrar.platformViewRegistry().registerViewFactory("com.airship.flutter/InboxMessageView", InboxMessageViewFactory(registrar))
         }
     }
 
+    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(binding.flutterEngine.dartExecutor, "com.airship.flutter/airship")
+        binding.platformViewRegistry.registerViewFactory("com.airship.flutter/InboxMessageView", InboxMessageViewFactory(binding))
+        channel.setMethodCallHandler(this)
+        context = binding.applicationContext
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
```

Comment:
should we also release the other references here (channel, context)?

Reply:
Yeah I released the channel here, but I don't think that releasing the context is needed.
However, I may be wrong